module/VuFind/src/VuFind/Recommend/LibGuidesProfile.php

```diff
@@ -134,8 +170,83 @@ public function process($results)
      */
     public function getResults()
     {
-        $query = $this->results->getParams()->getQuery();
-        $account = $this->findBestMatch($query);
+        // Consider strategies in the order listed in the config file.
+        foreach ($this->strategies as $strategy) {
+            if ("call_number" == $strategy) {
+                // Try the best match for call numbers in the results facets.
+                $account = $this->findBestMatchByCallNumbers($this->results);
+                if ($account) {
+                    return $account;
+                }
+            } elseif ("subject" == $strategy) {
+                // Find best match for the query terms alone.
+                $query = $this->results->getParams()->getQuery();
```

Comment:
That's fantastic.  I did eliminate $strategyMap and I added a method_exists($this) check, is that safe enough?

Reply:
I think it's pretty safe, though If you're feeling paranoid, you might also use a regex to strip weird characters off the strategy string to avoid shenanigans (not that I can think of any specific bad thing somebody could do) -- i.e. a `preg_replace('/[^\w]*/, '', $strategy)` type of thing.

Reply:
I'm not sure I'm wrapping my head around this, but if a method actually exists matching whatever the strategy string is, and it's somehow malicious, I'm not sure how restricting the method names to alphanumeric will necessarily help.  Couldn't the attacker just name their method something in the restricted character set?  I do get that we might want to exclude loading methods from other classes/paths somehow, but that seems to bring its own set of limitations.  I defer to your expertise here, you had the idea of dynamically determining the strategy method so I'm guessing similar approaches have been used elsewhere?

Reply:
Yeah fair point.  I added that sanitizing.